lib/client.py

```diff
@@ -54,56 +57,103 @@ def create(cls, config, username, password):
 
     def get_passwords(self, account, asset, gateway_username):
         with self.__requests_tls.open_session() as session:
+            passwords = []
             auth_token = self.__authenticator.authenticate(
                 session, self.__address_url, self.__username, self.__password
             )
-            passwords = self.__get_passwords(session, auth_token, account, asset, gateway_username)
+            found_objects = self.__search_for_account_asset_secrets(session, auth_token, account, asset)
+            logger.debug("Found the following secrets: %s", found_objects)
+            if len(found_objects) == 0:
+                logger.debug(
+                    "No password found in vault for this account and/or asset: account=%s, asset=%s", account, asset
+                )
+                self.__authenticator.logoff(session, self.__address_url)
+                return passwords
+            secret_endpoints = self.__create_secret_endpoints(self.__is_password, found_objects, account, asset)
+            logger.debug("Will retrieve secrets from the following endpoints: %s", ",".join(secret_endpoints))
+            for se in secret_endpoints:
+                logger.debug("Getting password from: %s", se)
+                passwords.append(self.__endpoint_extractor.extract_data_from_endpoint(
+                    session,
+                    endpoint_url=se,
+                    headers=self.__make_headers(auth_token),
+                    method="post",
+                    data=self.__contruct_post_data(gateway_username, asset),
+                ))
+            self.__authenticator.logoff(session, self.__address_url)
+            return passwords
+
+    def get_ssh_keys(self, account, asset, gateway_username):
+        print("ssh keys")
+        with self.__requests_tls.open_session() as session:
+            auth_token = self.__authenticator.authenticate(
+                session, self.__address_url, self.__username, self.__password
+            )
+            keys = []
+            found_objects = self.__search_for_account_asset_secrets(session, auth_token, account, asset)
+            if len(found_objects) == 0:
+                logger.debug(
+                    "No keys found in vault for this account and/or asset: account=%s, asset=%s", account, asset
+                )
+                self.__authenticator.logoff(session, self.__address_url)
+                return keys
+            secret_endpoints = self.__create_secret_endpoints(self.__is_ssh_key, found_objects, account, asset)
+            for se in secret_endpoints:
+                response = session.post(
+                    urljoin(self.__address_url, se),
+                    headers=self.__make_headers(auth_token),
+                    data=json.dumps(self.__contruct_post_data(gateway_username, asset)),
+                    params=None,
+                )
+                if response.ok:
+                    keys.append(response.text.strip("\""))
```

Comment:
Is the double " intended?

Reply:
Which one? We used double quotes everywhere hence I used it with the escape sequence

Reply:
Sorry I am not sure why I could not understand this. My bad.